supervision/detection/utils.py

```diff
@@ -20,7 +20,10 @@ def polygon_to_mask(polygon: np.ndarray, resolution_wh: Tuple[int, int]) -> np.n
     """
     width, height = resolution_wh
     mask = np.zeros((height, width))
-    cv2.fillPoly(mask, [polygon], color=1)
+
+    # Empty numpy array check
```

Comment:
Thank you for addressing the bug; I suggest using if polygon.size > 0: to directly check for an empty array, ensuring against cases with all-zero coordinates.